src/Controls/GlobeControls.js

```diff
@@ -250,16 +244,43 @@ class GlobeControls extends THREE.EventDispatcher {
         this._onKeyUp = this.onKeyUp.bind(this);
         this._onBlurListener = this.onBlurListener.bind(this);
 
+        this._onInitRotation = this.initRotation.bind(this);
+        this._onInitDrag = this.initDrag.bind(this);
+        this._onInitDolly = this.initDolly.bind(this);
+        this._onInitPan = this.initPan.bind(this);
+        this._onInitPanoramic = this.initPanoramic.bind(this);
+
+        this._onRotation = this.handleRotation.bind(this);
+        this._onDrag = this.handleDrag.bind(this);
+        this._onDolly = this.handleDolly.bind(this);
+        this._onPan = this.handlePan.bind(this);
+        this._onPanoramic = this.handlePanoramic.bind(this);
+
+        this._onEndMovement = this.handleEndMovement.bind(this);
+
+        this.states.addEventListener('rotate-init', this._onInitRotation, false);
+        this.states.addEventListener('drag-init', this._onInitDrag, false);
+        this.states.addEventListener('dolly-init', this._onInitDolly, false);
+        this.states.addEventListener('pan-init', this._onInitPan, false);
+        this.states.addEventListener('panoramic-init', this._onInitPanoramic, false);
```

Comment:
could you use one trigger event for state change?

Reply:
I added a 'state-changed' event to manage all initialization.